### ISSUE TITLE
[Suggestion] Rarity tier and roll% display for uniques

### ISSUE THREAD

**Issue by worldendDominator** [Lailloken/Lailloken-UI#342]
I suggest adding some stats to unique pop-up:
1) Rarity tier for global drop uniques (can be taken from PoE Ladder website).
2) Overall roll percentage, as sum of all rolls on the item against sums of lowest and highest rolls.

**Comment by Lailloken** [comment]
1. Thanks for bringing this to my attention, I've never heard of that site. Sounds like a neat addition.
2. I thought about adding "roll health" in the past because the bars in the background are sometimes misleading (depending on the type of stat and its roll-range). I put it into my backlog because I couldn't find a satisfactory solution that wouldn't clutter the tooltip (I wanted to include it for every mod, also on rares). I can look into it again and limit it to uniques for now because the tooltip has less information there.

**Comment by Lailloken** [comment]
I found some time today to play around with your suggestion, and here's the first draft:
![image](https://github.com/Lailloken/Lailloken-UI/assets/61888437/01a7ad40-e1f1-421d-9ae2-2d605bc78f91)
![image](https://github.com/Lailloken/Lailloken-UI/assets/61888437/02dd2bb9-3211-4e4f-a18e-5d064e896f1b)

Since it's your suggestion, feel free to change any of these aspects (or add anything):
- roll highlighting

  - perfect rolls are highlighted white + red

  - any other roll will be highlighted using the standard T1-T6 colors that can be customized in the settings: I don't know where to set the thresholds (for now it just goes down one tier every 5%, but that may be too strict)
  - the average roll-% uses the same highlighting as individual rolls

- rarity/drop-tier highlighting

  - T0 is highlighted white + red, the rest uses the colors mentioned above

  - non-numerical tiers simply use the `#` color, but those could be further distinguished if preferred

- %-values: could you check if the values make sense in the first screenshot? (especially the added flat dmg)

**Comment by worldendDominator** [comment]
What I had in mind was just the global percentage, not one for each stat. But if you think they could be useful, then sure.
Not specific to this addition, but do roll bars take into account when a stat has reversed benefit? Like increased damage taken on Abyssus, or -mana cost on Praxis.
It looks like final roll % is average of all stats' %s. What I suggested was sum of current rolls compared to sum of minimum rolls and sum of maximum rolls, mostly because that's the way PoELadder counts its roll rating and that's what I wanted to use it for.
Otherwise, I don't think there's anything wrong with those values.

**Comment by Lailloken** [comment]
> Not specific to this addition, but do roll bars take into account when a stat has reversed benefit? Like increased damage taken on Abyssus, or -mana cost on Praxis.

There are a few mods that I manually modified whenever I stumbled upon them and thought that a reversed bar would be more intuitive. There are also some inconsistencies within the game when it comes to whether the min or max value is better, e.g. the crusader mods "(51-60)% reduced ignite duration on you" and "(60-51)% reduced effect of shock on you".

> What I suggested was sum of current rolls compared to sum of minimum rolls and sum of maximum rolls, mostly because that's the way PoELadder counts its roll rating and that's what I wanted to use it for.

Do you have a screenshot of that? And how does PoE Ladder's rating work in general?

**Comment by worldendDominator** [comment]
>Do you have a screenshot of that? And how does PoE Ladder's rating work in general?
Copied from the explanation given to me:
1) Take the latest stat ranges for each variable item property
2) Translate it into a 1 to X value which represents the full range of values for the stat so a 10 - 20 range becomes 1 - 11
3) Add the max of all ranges for an item to give the potential total for a "perfectly rolled" item
4) Sum up the actual scores per stat and divide it by the number calculated in step 3 then multiply by 100 to give a %age

Essentially:
1) Add up the minimums of all variable stat ranges
2) Add up the maximums of all variable stat ranges
3) Add up actual rolls for all variable stats
4) Evaluate p.3 value's position on the scale of p.1 to p.2

**Comment by Lailloken** [comment]
Hey there, had to take a break because I got sick and then my dog had an injury. I'm probably going to pick things up again this weekend or so.
***
> do roll bars take into account when a stat has reversed benefit? Like increased damage taken on Abyssus, or -mana cost on Praxis.

I recently noticed that some rules I created broke at some point, which made me consider two things:
- just use the game's native min/max roll-values, which doesn't require coming up with (annoying) custom rules

- make the bars/values customizable on a mod-by-mod basis, so users can decide for themselves how they are displayed/calculated (off the top of my head, simply clicking a mod-text would invert the bars/values and save it for that specific string of text)

I'm leaning towards the second option but haven't checked how much work it would take to implement. An additional benefit would be that this approach is not limited to English game-clients (unlike custom rules).
***
> What I had in mind was just the global percentage, not one for each stat. But if you think they could be useful, then sure.

After playing with that version of the tooltip for a bit, I feel that individual %-values add a bit of glance-value to uniques (similar to how it does for rare items, so it somewhat streamlines usage between the two categories) -- especially those whose mod-pools are heavily weighted towards one or two specific mods in terms of being build-enabling.
